metrics/metrics_collector.go

```diff
@@ -0,0 +1,160 @@
+package metrics
+
+import (
+	"fmt"
+	"math"
+	"math/rand"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+type metricCollector struct {
+	counters   []prometheus.Counter
+	gauges     []prometheus.Gauge
+	histograms []prometheus.Histogram
+	summarys   []prometheus.Summary
+	interval   int
+}
+
+var (
+	promRegistry = prometheus.NewRegistry() // local Registry so we don't get Go metrics, etc.
+)
+
+func newMetricCollector() metricCollector {
+	return metricCollector{}
+}
+
+func (mc *metricCollector) updateCounter() {
+	for {
+		time.Sleep(time.Second * time.Duration(mc.interval))
+		for idx := 0; idx < len(mc.counters); idx++ {
+			mc.counters[idx].Add(rand.Float64())
+		}
+	}
+}
+
+func (mc *metricCollector) updateGauge() {
+	for {
+		time.Sleep(time.Second * time.Duration(mc.interval))
+		for idx := 0; idx < len(mc.gauges); idx++ {
+			mc.gauges[idx].Add(rand.Float64())
+		}
+	}
+}
+
+func (mc *metricCollector) updateHistogram() {
+	for {
+		time.Sleep(time.Second * time.Duration(mc.interval))
+
+		for idx := 0; idx < len(mc.histograms); idx++ {
+			lowerBound := math.Mod(rand.Float64(), 1)
+			increment := math.Mod(rand.Float64(), 0.05)
+			for i := lowerBound; i < 1; i += increment {
```

Comment:
This seems like it might introduce odd regularity to the generated distribution.  Perhaps it would be better to generate a random number of random values:

```go
count := rand.Intn(30) // some upper limit on the number of values to generate
for i := 0; i < count; i++ {
  mc.histograms[idx].Observe(rand.Float64()) // will be in the range [0.0,1.0)
}
```